src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -1796,6 +1806,28 @@ ep_rt_diagnostics_command_line_get (void)
 	return cmd_line;
 }
 
+static
+const ep_char16_t *
+ep_rt_entrypoint_assembly_path_get_ref_utf16 (void)
+{
+	// TODO: find mono entrypoint assembly path
+	return (const ep_char16_t *)L"TODO";
```

Comment:
mono_assembly_get_main will give you the assembly containing main method. Then you can get to the full path of the assembly using, m_image_get_filename (assembly->image). Please note that there might not be a main assembly in case where mono is used in embedding and no main method has been executed.

Reply:
NOTE that all mono functions uses UTF8, so you will need to convert to UTF16. If you want to handle this as a singleton you could do similar to the lazy init as already done with command line.

Reply:
I believe you could drop _ref_ in method names since it already returns const and that should be enough to indicate to caller that the string shouldn't be freed.

Reply:
> Please note that there might not be a main assembly in case where mono is used in embedding and no main method has been executed.

Any thoughts on what the value in this payload should be in this case since it's Mono specific?

Reply:
Sound like you could have the same scenario in coreclr so we should at least have the same fallback in case where there is no main assembly available. I believe just an empty string will be OK, but if we have some different placeholder for unknown in diagnostics library we should probably unify with that.